src/opnsense/mvc/app/library/OPNsense/Firewall/Plugin.php

```diff
@@ -233,16 +233,26 @@ public function anchorToText($types = "fw", $placement = "tail")
         foreach (explode(',', $types) as $type) {
             foreach ($this->anchors as $anchorKey => $anchor) {
                 if (strpos($anchorKey, "{$type}.{$placement}") === 0) {
-                    $result .= $type == "fw" ? "" : "{$type}-";
-                    $result .= "anchor \"{$anchor['name']}\"";
+                    $result .= ($type == "fw" || $type == "ether") ? "" : "{$type}-";
+                    $prefix = $type == "ether" ? "ether " : "";
+                    $result .= "{$prefix}anchor \"{$anchor['name']}\"";
                     if ($anchor['quick']) {
                         $result .= " quick";
                     }
+                    if (!empty($anchor['ifs'])) {
+                        $ifs = array_filter(array_map(function($if) {
+                            return $this->interfaceMapping[$if]['if'] ?? null;
+                        }, explode(',', $anchor['ifs'])));
+
+                        if (!empty($ifs)) {
+                            $result .= " on {" . implode(', ', $ifs) . "}";
+                        }
+                    }
                     $result .= "\n";
                 }
             }
         }
-        return $result;
+        return $result . "\n";
```

Comment:
Not sure why we need an extra line break here, if we want extra spacing for readability, it might be better to add the `\n` to the `$result` line above (`\n\n` instead of `\n`)